qutip/animation.py

```diff
@@ -0,0 +1,1447 @@
+"""
+Functions for creating animations of results of quantum dynamics simulations,
+visualizations of quantum states and processes.
+"""
+
+__all__ = ['anim_wigner_sphere', 'anim_hinton', 'anim_sphereplot',
+           'anim_matrix_histogram', 'anim_fock_distribution',
+           'anim_wigner', 'anim_spin_distribution', 'anim_qubism',
+           'plot_schmidt']
+
+import warnings
+import itertools as it
+import numpy as np
+from numpy import pi, array, sin, cos, angle, log2
+
+from packaging.version import parse as parse_version
+
+from . import (
+    Qobj, isket, ket2dm, tensor, vector_to_operator, to_super, settings
+)
+from .core.dimensions import flatten
+from .core.superop_reps import _to_superpauli, isqubitdims
+from .wigner import wigner
+from .matplotlib_utilities import complex_phase_cmap
+
+try:
+    import matplotlib.pyplot as plt
+    import matplotlib as mpl
+    from matplotlib import cm
+    from mpl_toolkits.mplot3d import Axes3D
+
+    # Define a custom _axes3D function based on the matplotlib version.
+    # The auto_add_to_figure keyword is new for matplotlib>=3.4.
+    if parse_version(mpl.__version__) >= parse_version('3.4'):
+        def _axes3D(fig, *args, **kwargs):
+            ax = Axes3D(fig, *args, auto_add_to_figure=False, **kwargs)
+            return fig.add_axes(ax)
+    else:
+        def _axes3D(*args, **kwargs):
+            return Axes3D(*args, **kwargs)
+except:
+    pass
+
+import os
+from functools import partial
+from IPython.display import HTML
+from base64 import b64encode
+import matplotlib.animation as animation
+from qutip import plot_wigner
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+from .visualization import *
+
+
+def make_html_video(ani, save_options=None):
+    if save_options is None:
+        save_options = dict()
+    if not isinstance(save_options, dict):
+        raise ValueError('save_options must be a dict')
+
+    if 'name' not in save_options.keys():
+        save_options['name'] = 'animation'
+    if 'writer' not in save_options.keys():
+        save_options['writer'] = None
+    if 'codec' not in save_options.keys():
+        save_options['codec'] = None
+
+    ani.save(save_options['name'] + '.mp4', fps=10,
+             writer=save_options['writer'], codec=save_options['codec'])
+    video = open(save_options['name'] + '.mp4', "rb").read()
+    video_encoded = b64encode(video).decode("ascii")
+    video_tag = '<video controls src="data:video/x-m4v;base64,{0}">'.format(
+        video_encoded)
+
+    return ani, HTML(video_tag)
```

Comment:
Using `%matplotlib notebook` I can show animation in a notebook without writing it to a file.
No `save`, thus no issue with codex and it's somewhat interactive as I can zoom in.
This function is therefore not needed. We can just return the animation object and let the user see it as is, or save it in a file if desired.

Reply:
Thanks for your info. I will change them to return fig and ani, but we need ```make_html_video```. In my case, I can use the interactive mode in one jupyter notebook, but not in the other and the google colab.